src/EnergyPlus/HybridUnitaryAirConditioners.cc

```diff
@@ -0,0 +1,728 @@
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+
+// C++ Headers
+#include <cassert>
+#include <cmath>
+
+// ObjexxFCL Headers
+#include <ObjexxFCL/Array.functions.hh>
+#include <ObjexxFCL/Fmath.hh>
+
+// EnergyPlus Headers
+#include <HybridUnitaryAirConditioners.hh>
+// EnergyPlus Headers
+#include <HybridEvapCoolingModel.hh>
+#include <DataEnvironment.hh>
+#include <BranchNodeConnections.hh>
+#include <CurveManager.hh>
+#include <DataAirSystems.hh>
+#include <DataContaminantBalance.hh>
+#include <DataEnvironment.hh>
+#include <DataGlobalConstants.hh>
+#include <DataHeatBalance.hh>
+#include <DataHeatBalFanSys.hh>
+#include <DataHVACGlobals.hh>
+#include <DataIPShortCuts.hh>
+#include <DataLoopNode.hh>
+#include <DataPrecisionGlobals.hh>
+#include <DataSizing.hh>
+#include <DataWater.hh>
+#include <DataZoneEnergyDemands.hh>
+#include <EMSManager.hh>
+#include <Fans.hh>
+#include <General.hh>
+#include <GeneralRoutines.hh>
+#include <InputProcessor.hh>
+#include <NodeInputManager.hh>
+#include <OutAirNodeManager.hh>
+#include <OutputProcessor.hh>
+#include <Psychrometrics.hh>
+#include <ReportSizingManager.hh>
+#include <ScheduleManager.hh>
+#include <UtilityRoutines.hh>
+#include <WaterManager.hh>
+
+//#include <HybridModelConfigFile.hh>
+
+#define  TEMP_CURVE 0 
+#define  W_CURVE 1
+#define  POWER_CURVE 2 
+
+namespace EnergyPlus {//***************
+
+	namespace HybridUnitaryAirConditioners {
+		// Using/Aliasing
+		using InputProcessor::FindItemInList;
+		using General::TrimSigDigits;
+		using HybridEvapCoolingModel::Model;
+
+		using CurveManager::GetCurveIndex;
+		using CurveManager::GetCurveType;
+		using CurveManager::GetCurveMinMaxValues;
+		using CurveManager::CurveValue;
+		using HybridEvapCoolingModel::CMode;
+
+		Array1D< Model > ZoneHybridUnitaryAirConditioner;
+		int NumZoneHybridEvap(0);
+		Array1D_bool CheckZoneHybridEvapName;
+		bool GetInputZoneHybridEvap(true);
+		//Begin routines for zone HVAC Hybrid Evaporative cooler unit
+		//_______________________________________________________________________________________________________________________
+		//***************
+		void
+			SimZoneHybridUnitaryAirConditioners(

```

Comment:
Still some formatting/indentation issues to address before we're all done, but this is looking much better overall.

Reply:
ok done